src/dune/expander.ml

```diff
@@ -409,24 +437,22 @@ let expand_and_record acc ~map_exe ~dep_kind ~expansion_kind
                path_exp path)
         in
         add_ddep dep
-    | Error e -> (
-      match lib_private with
-      | true -> add_fail acc { fail = (fun () -> raise e) }
-      | false ->
-        if Lib.DB.available (Scope.libs t.scope) lib then
-          let fail () =
-            raise
-              (User_error.raise ~loc
+    | Error e ->
+      add_fail
+        ( match lib_private with
+        | true -> e
+        | false ->
+          if Lib.DB.available (Scope.libs t.scope) lib then
+            User_error.E
+              (User_error.make ~loc
```

Comment:
This error is now created more eagerly than necessary. I don't think this is worth worrying about however.